### PR TITLE
fix(http): Destroy some got responses explicitly to avoid onCancel errors

### LINKED ISSUE #12127: unhandledRejection after 27.19.3
### How are you running Renovate?

Self-hosted

### Please select which platform you are using if self-hosting.

gitlab.com

### If you're self-hosting Renovate, tell us what version of Renovate you run.

27.19.3-27.31.10

### Describe the bug

The issue occurred already some versions before.
`[0KUsing Docker executor with image renovate/renovate:27.19.3-slim@sha256:166c38339487782f4cd9b263806cbddc1034a4f80cd6d8818904f492a3520677 ...[0;m
` was still ok

But on the next run with
`[0KUsing Docker executor with image renovate/renovate:27.19.3-slim@sha256:dc37e6d0e6c16398562b3e0df29703ed81876107d9268df841680659712326fc ...[0;m`
it stopped working.

There were no changes in any configuration, or I didn't see any, and it occurred out of nowhere, other than newer renovate docker images. Unfortunately there isn't more infos about the issue in the log. We have multiple gitlab repos and jobs for renovate and it occurs only on two of them, which makes it more confusing.

### Relevant debug logs

<details><summary>Logs</summary>

```
 INFO: Repository started (repository=xxx)
       "renovateVersion": "27.31.10"
 INFO: Dependency extraction complete (repository=xxx)
       "baseBranch": "master",
       "stats": {
         "managers": {
           "docker-compose": {"fileCount": 1, "depCount": 0},
           "dockerfile": {"fileCount": 8, "depCount": 8},
           "gitlabci": {"fileCount": 1, "depCount": 3},
           "gitlabci-include": {"fileCount": 1, "depCount": 1},
           "maven": {"fileCount": 158, "depCount": 431}
         },
         "total": {"fileCount": 169, "depCount": 443}
       }
ERROR: unhandledRejection (repository=xxx)
       "err": {
         "message": "The `onCancel` handler was attached after the promise settled.",
         "stack": "Error: The `onCancel` handler was attached after the promise settled.
             at onCancel (/usr/src/app/node_modules/p-cancelable/index.js:48:12)
             at makeRequest (/usr/src/app/node_modules/got/dist/source/as-promise/index.js:38:13)
             at Request.<anonymous> (/usr/src/app/node_modules/got/dist/source/as-promise/index.js:143:17)
             at Object.onceWrapper (events.js:520:26)
             at Request.emit (events.js:400:28)
             at Request.emit (domain.js:475:12)
             at Timeout.retry (/usr/src/app/node_modules/got/dist/source/core/index.js:1264:30)
             at listOnTimeout (internal/timers.js:557:17)
             at processTimers (internal/timers.js:500:7)"
       }
ERROR: unhandledRejection (repository=xxx)
       "err": {
         "message": "The `onCancel` handler was attached after the promise settled.",
         "stack": "Error: The `onCancel` handler was attached after the promise settled.
             at onCancel (/usr/src/app/node_modules/p-cancelable/index.js:48:12)
             at makeRequest (/usr/src/app/node_modules/got/dist/source/as-promise/index.js:38:13)
             at Request.<anonymous> (/usr/src/app/node_modules/got/dist/source/as-promise/index.js:143:17)
             at Object.onceWrapper (events.js:520:26)
             at Request.emit (events.js:400:28)
             at Request.emit (domain.js:475:12)
             at Timeout.retry (/usr/src/app/node_modules/got/dist/source/core/index.js:1264:30)
             at listOnTimeout (internal/timers.js:557:17)
             at processTimers (internal/timers.js:500:7)"
       }
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

### PR DESCRIPTION
**Pull request by zharinov**
## Changes:

Zero retries fixes `onClosed` error, though we can use it for certain problematic hosts.

## Context:

Closes #12127

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by rarkins — DISMISSED** [review]

**Comment by zharinov** [comment]
Still getting occasional errors, experimenting with more non-central Maven registries

File: lib/util/http/retry.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const skipRetryForTheseHosts = ['jcenter.bintray.com'];
+const skipRetryHostsSet = new Set(skipRetryForTheseHosts);
```
Comment (viceice) [inline lib/util/http/retry.ts:2]: Combine to one const.

File: lib/util/http/retry.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+export function skipRetry(url: string): boolean {
+  if (process.env.NODE_ENV === 'test') {
+    return true;
+  }
+
+  const { hostname } = new URL(url);
```
Comment (viceice) [inline lib/util/http/retry.ts:9]: Use our parseUrl function

**Review by viceice — CHANGES_REQUESTED** [review]
I think this needs new tests?

File: lib/util/http/hooks.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+// Renovate issue: https://github.com/renovatebot/renovate/issues/12127
+// Got issue: https://github.com/sindresorhus/got/issues/1489
+// From here: https://github.com/sindresorhus/got/issues/1489#issuecomment-805485731
```
Comment (zharinov) [inline lib/util/http/hooks.ts:3]: cc @rarkins @viceice

What do you think?

**Review by viceice — APPROVED** [review]

**Pull request merged**